runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingModeExecutionContext.java

```diff
@@ -150,9 +159,13 @@ public void start(
       @Nullable Instant synchronizedProcessingTime,
       WindmillStateReader stateReader,
       SideInputStateFetcher sideInputStateFetcher,
-      Windmill.WorkItemCommitRequest.Builder outputBuilder) {
+      Windmill.WorkItemCommitRequest.Builder outputBuilder,
+      @Nullable Supplier<Boolean> workFailed) {
     this.key = key;
     this.work = work;
+    if (workFailed != null) {
+      this.workIsFailed = workFailed;
+    }
```

Comment:
this shoudl clear the existing value in the else case, otherwise it could be examining previous work item failure state.

Reply:
I'll set it back to the default one that always returns `Boolean.FALSE` in that case. I think for all use cases, we either always supply a non-null  Supplier or a null one though.